sdks/python/apache_beam/utils/timestamp.py

```diff
@@ -140,6 +141,23 @@ def to_rfc3339(self):
     # Append 'Z' for UTC timezone.
     return self.to_utc_datetime().isoformat() + 'Z'
 
+  def to_proto(self):
+    """Returns the `google.protobuf.timestamp_pb2` representation."""
+    secs = self.micros // 1000000
+    nanos = (self.micros - (secs * 1000000)) * 1000
+    return timestamp_pb2.Timestamp(seconds=secs, nanos=nanos)
+
+  @staticmethod
+  def from_proto(timestamp_proto):
+    """Creates a Timestamp from a `google.protobuf.timestamp_pb2`.
+
+    Note that the google has a sub-second resolution of nanoseconds whereas this
+    class has a resolution of microsends. This class will truncate the
+    nanosecond resolution down to the microsecond.
+    """
+    return Timestamp(seconds=timestamp_proto.seconds,
+                     micros=timestamp_proto.nanos // 1000)
```

Comment:
I think we should fail here if we ever perform any rounding with a TODO stating that we should support nanos.

Reply:
Ack, it now raises a ValueError with a reference to a JIRA as well as a TODO comment.